homeassistant/components/alarm_control_panel/verisure.py

```diff
@@ -15,33 +17,39 @@
     STATE_UNKNOWN)
 
 _LOGGER = logging.getLogger(__name__)
+_STATUS = {}
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """ Sets up the Verisure platform. """
 
-    if not verisure.MY_PAGES:
-        _LOGGER.error('A connection has not been made to Verisure mypages.')
-        return False
-
     alarms = []
+    if int(verisure.HUB.config.get('alarm', '1')):
+        update_alarms()
+        alarms.extend([
+            VerisureAlarm(verisure.HUB.my_pages.alarm, value.id)
+            for value in _STATUS.values()
+            ])
+    add_devices(alarms)
 
-    alarms.extend([
-        VerisureAlarm(value)
-        for value in verisure.ALARM_STATUS.values()
-        if verisure.SHOW_ALARM
-        ])
 
-    add_devices(alarms)
+@Throttle(timedelta(seconds=1))
+def update_alarms():
```

Comment:
All these type of methods should also move to the VerisureHub class including converting the global dictionaries to be instance variables of the hub.

```python
get_alarm(self, id):
    update_alarms()
    return self._ALARMS[id]

@Throttle(…)
def update_alarms(self):
    self.update_component(…, self._ALARMS)
```